src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -33,162 +43,465 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
-        internal static DiagnosticDescriptor MethodTypeParameterMoreMeaningfulNameRule = new(RuleId,
-                                                                             s_localizableTitle,
-                                                                             s_localizableMessageMethodTypeParameterMoreMeaningfulName,
-                                                                             DiagnosticCategory.Naming,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1704",
-                                                                             customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);*/
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-        //ImmutableArray.Create(AssemblyRule, NamespaceRule, TypeRule, MemberRule, MemberParameterRule, DelegateParameterRule, TypeTypeParameterRule, MethodTypeParameterRule, AssemblyMoreMeaningfulNameRule, NamespaceMoreMeaningfulNameRule, TypeMoreMeaningfulNameRule, MemberMoreMeaningfulNameRule, MemberParameterMoreMeaningfulNameRule, DelegateParameterMoreMeaningfulNameRule, TypeTypeParameterMoreMeaningfulNameRule, MethodTypeParameterMoreMeaningfulNameRule);
-
-#pragma warning disable RS1025 // Configure generated code analysis
+        private static readonly SourceTextValueProvider<CodeAnalysisDictionary> _xmlDictionaryProvider = new SourceTextValueProvider<CodeAnalysisDictionary>(ParseXmlDictionary);
+        private static readonly SourceTextValueProvider<CodeAnalysisDictionary> _dicDictionaryProvider = new SourceTextValueProvider<CodeAnalysisDictionary>(ParseDicDictionary);
+        private static readonly CodeAnalysisDictionary _mainDictionary = GetMainDictionary();
+
+        internal static DiagnosticDescriptor FileParseRule = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessageFileParse,
+            DiagnosticCategory.Naming,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        internal static DiagnosticDescriptor AssemblyRule = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessageAssembly,
+            DiagnosticCategory.Naming,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false,
+            description: s_localizableDescription,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1704",
```

Comment:
I don't think the help link URL should be set manually.

Reply:
I think it is fine for rules in this experimental package. This may change in future.

Reply:
Hmm, already removed. Do you want them back?

Reply:
Yes, please set the help link as it already exists.

Reply:
I'm no longer able to after switching to DiagnosticDescriptorHelper.Create per [this comment[(https://github.com/dotnet/roslyn-analyzers/pull/1111#discussion_r556584821)

Reply:
Yes, the helper you used already passes a help link so you are all good.